app/src/main/java/openfoodfacts/github/scrachx/openfood/features/adapters/NutrimentsGridAdapter.kt

```diff
@@ -69,25 +64,40 @@ open class NutrimentsGridAdapter(
             private val vNutrimentServingValue: TextView = v.findViewById(R.id.nutriment_serving_value)
             private val vNutrimentValue: TextView = v.findViewById(R.id.nutriment_value)
 
-            fun fillNutrimentValue(item: NutrimentListItem) {
+            fun fillNutriment(item: NutrimentListItem) {
+                fillNutrimentName(item)
+                fillNutrimentValue(item)
+                fillServingValue(item)
+            }
+
+
+            private fun fillNutrimentName(item: NutrimentListItem) {
                 vNutrimentName.text = item.title
-                vNutrimentValue.append("${item.modifier} ${item.value} ${item.unit}")
             }
 
-            fun fillServingValue(item: NutrimentListItem) {
-                val servingValue = item.servingValue
-                if (servingValue.isBlank()) {
+            private fun fillNutrimentValue(item: NutrimentListItem) {
+                val value = item.value
+                if (value == null) {
+                    vNutrimentValue.visibility = View.INVISIBLE
+                } else {
+                    vNutrimentValue.text = "${item.modifierStr} $value ${item.unitStr}"
```

Comment:
should we set visibility to visible here?

Reply:
Yeah we can, despite it is already set in the XML.